test/test_pid_reuse.c

```diff
@@ -0,0 +1,122 @@
+/* Test if children processes can reuse PID while NX window is open. */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include <assert.h>
+#include <errno.h>
+#include <pthread.h>
+#include "zlib.h"
+
+/* globals shared by all threads */
+z_stream c_stream;
+
+void* comp_file_multith (void *iterations)
+{
+        int err;
+        long *sleep_sec;
+
+	pid_t my_tid = syscall(SYS_gettid);
+        fprintf(stderr, "thread started: pid: %d\n", (int) my_tid);
+
+        /* open the window */
+	err = deflateInit(&c_stream, Z_DEFAULT_COMPRESSION);
+        if (err != 0) {
+                printf("nx_deflateInit err %d\n", err);
+                return (void *) 1;
+        }
+
+	sleep_sec = (long *) iterations;
+	sleep(sleep_sec[0]);
+
+	fprintf(stderr, "thread %d exits\n", (int) my_tid);
+	return (void *) 0;
+}
+
+void fork_children (int parent_pid) {
+	int child_id, i = 0;
+	int seconds = 10;
+	int children = 2048;
+
+	for (i=0; i < children; i++) {
+		child_id = fork();
+
+		if (child_id > 0) {
+			if (child_id == parent_pid) {
+				fprintf(stderr, "!!! pid %d reused (see ps)\n",
+					child_id);
+				sleep(10000);
```

Comment:
Why is this sleep necessary?
That's 10000 seconds.  Why not exiting?

Reply:
To keep this child alive until the point we get the csb error signal.